app-server/app/src/main/scala/bucket/Filter.scala

```diff
@@ -0,0 +1,85 @@
+package com.azavea.rf.bucket
+
+import com.lonelyplanet.akka.http.extensions.{PageRequest, Order}
+
+import com.azavea.rf.datamodel.driver.ExtendedPostgresDriver
+import com.azavea.rf.datamodel.latest.schema.tables._
+import com.azavea.rf.utils.queryparams._
+
+
+object BucketFilters {
+
+  import ExtendedPostgresDriver.api._
+
+  type BucketsQuery = Query[Buckets, Buckets#TableElementType, Seq]
+
+  implicit class BucketDefault[M, U, C[_]](buckets: BucketsQuery) {
+
+    def filterByOrganization(orgParams: OrgQueryParameters): BucketsQuery = {
+      buckets.filter{
+        bucket => orgParams.organization.map(bucket.organizationId === _)
+          .reduceLeftOption(_ || _).getOrElse(true:Rep[Boolean])
+      }
+    }
+
+    def filterByUser(userParams: UserQueryParameters): BucketsQuery = {
+      buckets.filter{
+        bucket => List(
+          userParams.createdBy.map(bucket.createdBy === _),
+          userParams.modifiedBy.map(bucket.modifiedBy === _)
+        ).collect({case Some(criteria)  => criteria}).reduceLeftOption(_ && _).getOrElse(true:Rep[Boolean])
+      }
+    }
+
+    def filterByTimestamp(timeParams: TimestampQueryParameters): BucketsQuery = {
+      buckets.filter{ bucket =>
+        List(
+          timeParams.minCreateDatetime.map(bucket.createdAt > _),
+          timeParams.maxCreateDatetime.map(bucket.createdAt < _),
+          timeParams.minModifiedDatetime.map(bucket.modifiedAt > _),
+          timeParams.maxModifiedDatetime.map(bucket.modifiedAt < _)
+        ).collect({case Some(criteria)  => criteria}).reduceLeftOption(_ && _).getOrElse(true:Rep[Boolean])
+      }
+    }
+
+    def sort(sortMap: Map[String, Order]): BucketsQuery = {
```

Comment:
Sorting doesn't occur if they aren't provided.

It looks like based on the source of the [library](https://github.com/lonelyplanet/akka-http-extensions/blob/master/src/main/scala/com/lonelyplanet/akka/http/extensions/PaginationDirectives.scala#L88) we're using the parameters get dropped. I tried adding a case statement to catch when it is omitted and it never got to it.